### PR TITLE
New Flowhub login

### PR DESCRIPTION
**Pull request by bergie**
This PR changes login from The Grid Passport to direct GitHub authentication. Pending server updates.

README documents how to set up GitHub OAuth for custom URLs.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 NoFlo UI ChangeLog
 ==================
 
+# 0.14.0 (git master)
+
+Breaking changes
+
+* Switched from The Grid to direct GitHub authentication. This means changes are required for running NoFlo UI in custom URLs. See the README for more information
```
Comment (jonnor) [inline CHANGES.md]: "in custom URLs"? I guess this mean "when self-hosted"?

File: Gruntfile.coffee [hunk 2]
```diff
@@ -113,16 +113,19 @@ module.exports = ->
             replacement: process.env.NOFLO_OAUTH_SERVICE_USER or 'https://api.flowhub.io'
           ,
             pattern: /\$NOFLO_OAUTH_CLIENT_ID/ig
-            replacement: process.env.NOFLO_OAUTH_CLIENT_ID or '9d963a3d-8b6f-42fe-bb36-6fccecd039af'
+            replacement: process.env.NOFLO_OAUTH_CLIENT_ID or '46fe25abef8d07e6dc2d'
+          ,
+            pattern: /\$NOFLO_OAUTH_CLIENT_REDIRECT/ig
+            replacement: process.env.NOFLO_OAUTH_CLIENT_REDIRECT or 'http://localhost:9999'
```
Comment (jonnor) [inline Gruntfile.coffee]: Can you add a comment what this is for? Something like. `# Hosting location to redirect to after OAuth` ?

File: Gruntfile.coffee [hunk 3]
```diff
@@ -331,20 +334,13 @@ module.exports = ->
           urls: ['http://127.0.0.1:9999/spec/runner.html']
           browsers: [
             browserName: 'googlechrome'
-            platform: 'OS X 10.8'
```
Comment (jonnor) [inline Gruntfile.coffee]: Why remove? And this seems quite unrelated...

Comment (bergie) [inline Gruntfile.coffee]: Yes, unrelated

File: README.md [hunk 4]
```diff
@@ -71,10 +71,33 @@ If you prefer, you can also start a watcher process that will do a rebuild whene
...
+    $ ./node_modules/.bin/simple-server . 9999
 
-Where 3005 is the port you want the server to run. Once it is built and the server is running you can access the UI at `http://localhost:3005/index.html`
+Where 9999 is the port you want the server to run. Once it is built and the server is running you can access the UI at `http://localhost:9999/index.html`
 
 In addition to this project, the other repository of interest is the [the-graph](https://github.com/the-grid/the-graph) graph editor widget used for editing flows.
 
+## Authentication and custom URLs
 
+NoFlo UI is using GitHub for authentication. We have a default application configured to work at `http://localhost:9999`. If you want to serve your NoFlo UI from a different URL, you need to register your own [OAuth application](https://github.com/settings/applications/new) with them. Make sure to match GitHub's [redirect URL policy](https://developer.github.com/v3/oauth/#redirect-urls).
```
Comment (jonnor) [inline README.md]: Wii docs <3

File: components/RemoteLogin.coffee [hunk 5]
```diff
@@ -108,7 +127,13 @@ exports.getComponent = ->
 
     # With browser we check the URL also initially
     checkToken window.location.href, c.params, (err, token) ->
-      return if err
+      if err
+        # Send error with timeout so other context-affecting ops have time to finish
```
Comment (jonnor) [inline components/RemoteLogin.coffee]: What would happen if we did not do this?

Comment (bergie) [inline components/RemoteLogin.coffee]: @jonnor right after there is a context update from another store that "OKs" the state, causing the error message flash very briefly
